edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/DeviceMessageHandler.cs

```diff
@@ -102,7 +102,7 @@ public async Task ProcessMessageFeedbackAsync(string messageId, FeedbackStatus f
                 }
                 else
                 {
-                    taskCompletionSource.SetException(new EdgeHubIOException($"Message not completed by client {this.Identity.Id}"));
+                    taskCompletionSource.SetException(new EdgeHubMessageRejectedException($"Message not completed by client {this.Identity.Id}"));
```

Comment:
is it possible that feedback status is aborted here? if so, why throw Rejected Exception?

Reply:
We don't support Aborted. Plus the idea here is to not throw an IOException

Reply:
And if new exception is thrown here, this message will not be retried, why not retry?

Reply:
If the message is rejected, then it should not be retried. Else we will get into a loop of retrying the same message and it being rejected.